docs/modules/ROOT/pages/how-tos/configuring/creating-secrets.adoc

```diff
@@ -2,18 +2,23 @@
 
 When you build your pipeline, you might want to add tasks that require **secrets** in order to access external resources.
 
-NOTE: One such task is the link:https://github.com/konflux-ci/build-definitions/tree/main/task/sast-snyk-check[sast-snyk-check] task that uses the third-party service link:https://snyk.io/[snyk] to perform static application security testing (SAST) as a part of the default {ProductName} pipeline. Use this procedure to upload your snyk.io token. Name the secret `sast_snyk_task` so that the snyk task in the {ProductName} pipeline will recognize it and use it.
+.Secrets fall into three catagories of when they can be added:
+
+. Before a component is added. If a secret is needed to access the source control platform (like xref:creating-secrets.adoc#creating-source-control-secrets[GitLab]), then the secret must be created before the component is created.
+. Before a build succeeds. Some artifact build tasks need specific secrets to be able to pull all of the content to include in the final artifact. These secrets like for xref:creating-secrets.adoc#creating-registry-pull-secrets[container registries] can be added after the component is created but must be provided before a successful build can occur.
```

Comment:
```suggestion
. Before a build succeeds. Some artifact build tasks need specific secrets to be able to pull all of the content to include in the final artifact. For example, you can add secrets for xref:creating-secrets.adoc#creating-registry-pull-secrets[container registries] after you create the component but they must be provided before a successful build can occur.
```
Same as on line 8, you can go for a more user-focused 'you'. And such texts are faster and easier to read than the ones with the passive voice all around.